### PR TITLE
[REVIEW] Fix some docs build warnings

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR fixes some docstrings and add's references to `numpy.dtype` at some places.

### CODE REVIEW

File: docs/cudf/source/conf.py [hunk 1]
```diff
@@ -235,10 +237,14 @@ def resolve_aliases(app, doctree):
 
 def ignore_internal_references(app, env, node, contnode):
     name = node.get("reftarget", None)
-    if name is not None and name in _internal_names_to_ignore:
+    if name == "cudf.core.index.GenericIndex":
```
Comment (vyasr) [inline docs/cudf/source/conf.py:240]: Could you add a comment explaining why this is necessary? The internals of Sphinx node modifications are pretty arcane and won't be familiar to most people.

**Review by vyasr — APPROVED** [review]
One request, otherwise LGTM.

**Review by bdice — APPROVED** [review]
Ooh, that fix for `GenericIndex` probably reduces a lot of warnings! Nice job.

File: docs/cudf/source/conf.py (hunk 1 above)
Comment (galipremsagar) [reply docs/cudf/source/conf.py:240]: Done

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**